src/main/java/com/mongodb/hibernate/internal/translate/mongoast/command/aggregate/stage/AstProjectStageSpecification.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.internal.translate.mongoast.command.aggregate.stage;
+
+import com.mongodb.hibernate.internal.translate.mongoast.AstNode;
+import org.bson.BsonWriter;
+
+public abstract class AstProjectStageSpecification implements AstNode {
+
+    public static AstProjectStageSpecification include(String path) {
+        return new AstProjectStageSpecification() {
```

Comment:
Why are we using here an approach different from the one we use when modelling the rest of the AST?

- We do not have an abstract class `AstCommand`, with its static methods defining its concrete extensions and instantiating them. Instead, we have the `AstCommand` interface with multiple records or classes implementing it.
- The same approach as with `AstCommand` is used for `AstFilter`, `AstFilterOperation`, `AstStage`.

It seems to me, we should use the same approach here:

- Introduce the `AstProjectStageSpecification` interface, with a single (for now) `AstProjectStageIncludeSpecification` implementation.

Reply:
We defintively could do that. The only reason I chose this way is to accommodate many potential project stage specifications in the future, all of which involves simple logic and putting them together (as done in C# Driver; actually all the specifications are put into AstProjectStage for only AstProjectStage uses them) avoids clutter and distraction.

Do you think it is even better to move them into `AstProjectStage` class?

Reply:
refactored. a testing case is also added.